File: Source/Fantasy.Jobs/Scheduling/IScheduleLibraryService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace Fantasy.Jobs.Scheduling
{
    [ServiceContract(Namespace=Consts.ScheduleNamespaceURI)]
    public interface IScheduleLibraryService
    {
        [OperationContract] 
        string[] GetGroups(string path);

        [OperationContract] 
        void CreateGroup(string path);

        [OperationContract] 
        void DeleteGroup(string path);

        [OperationContract] 
        void CreateSchedule(string path, ScheduleItem schedule, bool overwrite);

        [OperationContract] 
        void DeleteSchedule(string path);

        [OperationContract] 
        string[] GetScheduleNames(string path);

        [OperationContract] 
        ScheduleItem GetSchedule(string path);

        [OperationContract] 
        ScheduleEvent[] GetScheduleHistory(string path);

        [OperationContract]
        string[] GetTemplateNames();

        [OperationContract]
        string GetTemplate(string name);

    }
}

File: Fantasy.Jobs/src/fantasy/jobs/scheduling/IScheduleLibraryService.java
package fantasy.jobs.scheduling;

import java.rmi.Remote;

public interface IScheduleLibraryService extends Remote
{

	String[] getGroups(String path) throws Exception;

	void createGroup(String path) throws Exception;

	void deleteGroup(String path) throws Exception;

	void createSchedule(String path, ScheduleItem schedule, boolean overwrite) throws Exception;

	void deleteSchedule(String path) throws Exception;

	String[] getScheduleNames(String path) throws Exception;


	ScheduleItem getSchedule(String path) throws Exception;

	ScheduleEvent[] getScheduleHistory(String path) throws Exception;

	String[] getTemplateNames() throws Exception;

	String getTemplate(String name) throws Exception;

}